lib/patterns_targets.py

```diff
@@ -19,81 +56,129 @@ def update_recursive(d, u):
 
 class SeqConfig(object):
     def __init__(self, config, patterns):
+        """
+         This class takes care of common tasks related to config and patterns
+         files (reading the sampletable, etc) but is intended to be subclassed.
 
+        Parameters
+        ----------
+        config : str or dict
+
+        patterns : str
+            Path to patterns YAML file
+        """
         self.path = None
         if isinstance(config, str):
             self.path = config
 
-        configdict, pth = common.resolve_config(config)
-        self.configdict = configdict
+        self.config = common.resolve_config(config)
 
-        common.get_references_dir(config)
+        # Read the config file and extract all sort of useful bits. This mostly
+        # uses the `common` module to handle the details.
+        config['references_dir'] = common.get_references_dir(config)
         self.samples, self.sampletable = common.get_sampletable(config)
         self.refdict, self.conversion_kwargs = common.references_dict(config)
-
-        self.assembly = configdict['assembly']
-
-        self.patterns_yaml = patterns
+        self.assembly = self.config['assembly']
+        self.patterns = yaml.load(open(patterns))
 
 
 class RNASeqConfig(SeqConfig):
-    def __init__(self, config, patterns=None):
-        if patterns is None:
-            patterns = os.path.join(HERE, 'rnaseq_patterns.yaml')
+    def __init__(self, config, patterns):
+        """
+        Config object specific to RNA-seq workflows.
+        """
         SeqConfig.__init__(self, config, patterns)
-
-        self.sample_dir = self.configdict.get('sample_dir', 'samples')
-        self.agg_dir = self.configdict.get('aggregation_dir', 'aggregation')
-
-        self.fill = dict(sample=self.samples, sample_dir=self.sample_dir, agg_dir=self.agg_dir)
+        self.sample_dir = self.config.get('sample_dir', 'samples')
+        self.agg_dir = self.config.get('aggregation_dir', 'aggregation')
+        self.fill = dict(sample=self.samples, sample_dir=self.sample_dir,
+                         agg_dir=self.agg_dir)
```

Comment:
Would it be better to use the sample table to fill instead of a fill dict? 

While the fill dict is more explicit, using the sample table is more flexible. We could add arbitrary fill patterns to the YAML and fill them with arbitrary values in the sample table.

Reply:
Or I guess if we want custom fills, we could just subclass and make modifications we would need.

Reply:
For that to happen we'd have to attach `sample_dir` and `agg_dir` as columns to the sampletable, right?